eval.go

```diff
@@ -98,6 +99,15 @@ type scope struct {
 	blocks    map[string]*BlockNode
 }
 
+func (s scope) sortedBlocks() []string {
```

Comment:
not sure sorting blocks by name is actually helpful. the most useful way to present them in the dump would be in the order they were defined in, but that's impossible, so just printing in random order is okay IMO.

Reply:
I agree that the order in which blocks were created would probably be the most helpful, however disagree with the notion of random ordering. 

Perhaps it is better to get predictable ordering, so that if you make a subtle change to the template, and compare differences between current rendering and previous rendering, you do not have to bother yourself with differences that are not meaningful?

Reply:
Yes, I can see the benefit there. Let's go with that!